src/snowflake/snowpark/session.py

```diff
@@ -634,6 +615,21 @@ def __init__(
 
         _logger.info("Snowpark Session information: %s", self._session_info)
 
+        # Register self._close_at_exit so it will be called at interpreter shutdown
+        atexit.register(self._close_at_exit)
+
```

Comment:
I don't understand how this fixes that order. can you elaborate on this.

Reply:
In connector, the exit function is registered in `__init__`. https://github.com/snowflakedb/snowflake-connector-python/blob/3c8255935b9aebc2edee8de3913959f86596739f/src/snowflake/connector/connection.py#L463. In snowpark, the exit function was originally registered at module level, so it happen before connector. After moving it into `__init__`, it will happen after connector.